syntax = "proto3";

package avatar_sync;

import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";

service AvatarSyncService {
  rpc SyncAvatarStream(stream AvatarSyncStreamRequest) returns (google.protobuf.Empty);
}

message AvatarSyncStreamRequest {
  oneof request_data {
    SyncConfig config = 1;
    bytes audio_chunk = 2;
    VisemeData viseme_data = 3;
  }
}

message SyncConfig {
  // string session_id = 1; // 기존 session_id는 frontend_session_id로 대체하거나,
                           // AvatarSync 내부용으로 별도 관리할 수 있습니다.
                           // 여기서는 frontend_session_id를 핵심 ID로 사용합니다.
  string frontend_session_id = 1; // ★ 필드 번호 1로 변경 또는 새로 추가하고 session_id는 제거/주석처리
  // string avatar_model_id = 2;
}

message VisemeData {
  string viseme_id = 1;
  google.protobuf.Timestamp start_time = 2;
  float duration_sec = 3;
}